### PR TITLE
Add noop prover that always returns unknown.

### PR DESCRIPTION
**Pull request by keyboardDrummer**
### Changes

Add a 'noop' prover that always returns unknown. Useful for getting prover logs quickly without having to wait for a solver

Can be turned on by passing `-proverOpt:SOLVER=noop`. Updated documentation to include this.

### Testing

Added a test that uses the 'noop' solver.

### CODE REVIEW

File: Source/Provers/SMTLib/FunctionDependencyCollector.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
+using System.Collections.Generic;
+using Microsoft.Boogie.VCExprAST;
+
+namespace Microsoft.Boogie.SMTLib;
+
+public class FunctionDependencyCollector : BoundVarTraversingVCExprVisitor<bool, bool>
```
Comment (keyboardDrummer) [inline Source/Provers/SMTLib/FunctionDependencyCollector.cs:6]: Class moved into its own file, but is unchanged.

File: Source/Provers/SMTLib/SMTLibProcessTheoremProver.cs [hunk 2]
```diff
@@ -11,35 +11,6 @@
 
 namespace Microsoft.Boogie.SMTLib
 {
-  public class FunctionDependencyCollector : BoundVarTraversingVCExprVisitor<bool, bool>
```
Comment (keyboardDrummer) [inline Source/Provers/SMTLib/SMTLibProcessTheoremProver.cs:14]: Class was moved to another file

File: Source/Provers/SMTLib/SMTLibProverOptions.cs [hunk 3]
```diff
@@ -102,7 +106,7 @@ protected override bool Parse(string opt)
             Solver = SolverKind.YICES2;
             break;
           default:
-            ReportError("Invalid SOLVER value; must be 'Z3' or 'CVC5' or 'Yices2'");
+            ReportError("Invalid SOLVER value; must be 'Z3' or 'CVC5', 'Yices2' or 'noop'");
```
Comment (RustanLeino) [inline Source/Provers/SMTLib/SMTLibProverOptions.cs:109]:
```suggestion
            ReportError("Invalid SOLVER value; must be 'Z3' or 'CVC5' or 'Yices2' or 'noop'");
```

**Review by RustanLeino — APPROVED** [review]
I like it! Good option to have.

**Review by RustanLeino — APPROVED** [review]

**Pull request merged**